crates/sui-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -96,12 +121,17 @@ impl SharedObjectCongestionTracker {
         Some((deferral_key, congested_objects))
     }
 
-    pub fn bump_object_execution_cost(
-        &mut self,
-        shared_input_objects: &[SharedInputObject],
-        tx_cost: u64,
-    ) {
-        let start_cost = self.compute_tx_start_at_cost(shared_input_objects);
+    // Update shared objects' execution cost used in `cert` using `cert`'s execution cost.
+    // This is called when `cert` is scheduled for execution.
+    pub fn bump_object_execution_cost(&mut self, cert: &VerifiedExecutableTransaction) {
+        let tx_cost = match self.mode {
+            PerObjectCongestionControlMode::None => return,
+            PerObjectCongestionControlMode::TotalGasBudget => cert.gas_budget(),
+            PerObjectCongestionControlMode::TotalTxCount => 1,
+        };
```

Comment:
can this be factored out with the duplicated code above?

Reply:
The `PerObjectCongestionControlMode::None` was different so I didn't merge them before. But it's probably a good idea to dedup them. Code updated.